mesoscope/cellpack
# Analysis tools to compare packings

Problem
=======
We need to add analysis tools to compare between different packings

Solution
========
Create functions to compare packings using x, y, z positions, as well as simplified parametrized representations for packings inside intracellular space.

## Type of change

* New feature (non-breaking change which adds functionality)
* Breaking change (fix or feature that would cause existing functionality to not work as expected)
* This change requires a documentation update
* This change requires updated or new tests

Change summary:
---------------
Added functions to:

1. calculate the similarity score using the Anderson-Darling test and plot heatmaps to compare various packings
1. create parametrized representations for the packings, calculate the correlation between packings, and create heatmaps for comparison

also closes #97 

Steps to Verify:
----------------
1. run `pack -r cellpack/test-recipes/v2/peroxisomes.json -c packing-configs/analyze_pex_packing.json` to create output files for 10 instances of peroxisome packings in a grid with spacing 4. Grid creation may take some time during the first iteration, but subsequent packing should load the grid from a saved file. 
1. run `analyze -i out/analyze/peroxisomes_test/spheresSST/ -o out/analyze/peroxisomes_test/` to run analysis tools on the packing data. Changing options for analysis tools is currently unsupported but will be added soon. 

Screenshots (optional):
-----------------------
Example output comparing two different packing methods:
![image](https://user-images.githubusercontent.com/101426188/195944657-693570fd-2c33-4dd2-8eb0-526a2caa7566.png)


Keyfiles (delete if not relevant):
-----------------------
1. `cellpack/bin/analyze.py`
3. `cellpack/autopack/Analysis.py`
